src/hooks/useDebouncedFetchSections.ts

```diff
@@ -78,19 +78,25 @@ const useDebouncedFetchSection = (
   }
 
   useEffect(() => {
-    if (debouncedSearchTerm) {
-      cioClient?.autocomplete
-        .getAutocompleteResults(debouncedSearchTerm, autocompleteParameters)
-        .then((response) => {
+    (async () => {
+      if (debouncedSearchTerm) {
+        try {
+          const response = await cioClient?.autocomplete.getAutocompleteResults(
+            debouncedSearchTerm,
+            autocompleteParameters
+          );
           const newSectionsData = transformResponse(response, {
             numTermsWithGroupSuggestions,
             numGroupsSuggestedPerTerm,
           });
           setSectionsData(newSectionsData);
-        });
-    } else if (!debouncedSearchTerm) {
-      setSectionsData({});
-    }
+        } catch (error: any) {
+          throw new Error(error);
```

Comment:
😕 how does this pattern help us here:
1. catch the OG error
2. create a new error based on the OG error
3. then throw our new error

The three steps above are how expect this new code to behave. Does that match your understanding? If not, what am I missing here? If so, how is this any different than not catching the OG error?